astropy/table/typing.py

```diff
@@ -0,0 +1,15 @@
+from typing import Protocol
+
+from astropy.utils.data_info import MixinInfo
+
+__all__ = [
+    "MixinColumn",
+]
+
+
+class MixinColumn(Protocol):
```

Comment:
Can downstream packages use the `Protocol` even if we don't have the [`py.typed` marker](https://peps.python.org/pep-0561/#packaging-type-information) yet?

Reply:
I guess it depends on what use you have in mind. IDEs should be happy to grab it even without py.typed, and last I checked that's been our use of type hints ?